### PR TITLE
Add N2N channel key rotation

### PR DESCRIPTION
**Pull request by wintersteiger**
This adds the facilities to trigger key exchanges for node-to-node channels when the message limits have been reached. This is item 3 of #2092.

### CODE REVIEW

**Comment by wintersteiger** [comment]
In debug mode, the first CI run used a very small message limit, i.e. a new key exchange is performed every 25 messages. This was just for the first test to see whether I can trigger any errors or assertions, but I couldn't. I'll raise that limit now so that there's no impact on performance.

File: src/node/channels.h [hunk 1]
```diff
@@ -631,24 +681,29 @@ namespace ccf
...
+      if (status == WAITING_FOR_FINAL)
         return;
 
       LOG_INFO_FMT("Initiating node channel with {}.", peer_id);
 
-      status = INITIATED;
+      key_exchange_in_progress = true;
+
+      if (status != ESTABLISHED)
+        status = INITIATED;
```
Comment (achamayou) [inline src/node/channels.h:696]:
```suggestion
      }
        status = INITIATED;
      }
```

**Review by achamayou — APPROVED** [review]

File: src/node/channels.h (hunk 1 above)
Comment (wintersteiger) [reply src/node/channels.h:696]: Awww, I thought I caught all of them this time!

File: src/node/test/channels.cpp [hunk 2]
```diff
@@ -501,20 +514,20 @@ TEST_CASE("Replay and out-of-order")
 
   INFO("Restart, message, resume");
```
Comment (jumaffre) [inline src/node/test/channels.cpp:515]: Please correct me if I'm wrong but I don't think we test for the auto-refreshing of the channel when the message limit has been reached. If the message limit is passed as an argument to the `Channel` constructor, this should be easy to set it to a low value in this unit test and see if the `key_exchange_init` message is produced.

File: src/node/channels.h [hunk 3]
```diff
@@ -23,6 +23,13 @@ namespace ccf
   using SeqNo = uint64_t;
   using GcmHdr = crypto::GcmHeader<sizeof(SeqNo)>;
 
+#ifndef NDEBUG
+  static constexpr size_t message_limit = 2048;
```
Comment (jumaffre) [inline src/node/channels.h:27]: Perhaps add a comment here to explain why we do this (i.e. see a few refresh during our perf test)?

**Review by jumaffre — APPROVED** [review]

File: src/node/test/channels.cpp (hunk 2 above)
Comment (wintersteiger) [reply src/node/test/channels.cpp:515]: Good point, I'll add a new test for that!

**Pull request merged**